connectors-home/modules/ROOT/pages/introduction/intro-config-oauth2.adoc

```diff
@@ -0,0 +1,189 @@
+
+. Configure the OAuth 2.0 callback configuration. Create a new *HTTP Listener* configuration and set *Protocol* to `HTTP (Default)`, *Host* to `localhost`, and *Port* to `8080`.
++
+image::intro-config-oauth2-gmail-http-listener.png[HTTP Listener Configuration window]
++
+Here is the XML for this *HTTP Listener*:
++
+[source,xml,linenums]
+----
+<http:listener-config name="HTTP_Listener_config" >
+		<http:listener-connection host="localhost" port="8080" />
+</http:listener-config>
+----
++
+. Select this *HTTP Listener* in the Gmail Connector connection configuration.
+. Set *Callback path* to `callback` and *Authorize path* to `authorize`. You can leave the *External Callback Url* field empty.
++
+image::intro-config-oauth2-gmail-callback.png[Gmail OAuth 2.0 Callback Configuration window]
++
+Here is the XML for the OAuth 2.0 callback configuration for Gmail Connector:
++
+[source,xml,linenums]
+----
+<gmail:config name="Gmail_Connector_Config" >
+    <gmail:oauth2c-connection >
+        <gmail:oauth-authorization-code consumerKey="YOUR_CLIENT_ID" consumerSecret="YOUR_CLIENT_SECRET" />
+        <gmail:oauth-callback-config listenerConfig="HTTP_Listener_config" callbackPath="callback" authorizePath="authorize"/>
+    </gmail:oauth2c-connection>
+</gmail:config>
+----
++
+. Run the application.
+
+=== Performing the OAuth Dance
+
+Perform the OAuth dance and authorize your application to access your Gmail account via the connector. The following steps show how to perform the OAuth dance:
+
+. In your browser, navigate to the path specified in the *Authorize Path* field to begin the OAuth dance. In this example, the path is `\http://localhost:8080/authorize`. If everything is configured correctly, you will see a Google Authentication screen in which you select your user account and give permission to the app to access your account. Google shows a list of permissions that require approval, for example:
++
+image::intro-config-oauth2-gmail-permissions.png[Window that asks the user permission for Gmail to access their account]
++
+NOTE: These permissions are dependent on the *Scopes* field configured in the Gmail Connector connection configuration. You can accept the defaults, but if you want to limit the access, you can remove the unwanted scopes. For example, remove the `\https://www.googleapis.com/auth/gmail.send` scope to disallow the connector from sending new emails from your account, however, doing so also limits connector functionality, such as the *Send Message* or the *Send Draft* operations.
++
+. Click *Allow*. You are now redirected to `\http://127.0.0.1:8080/callback`, which is the redirect URI that you configured in the OAuth provider and is open in the connector configuration. If the OAuth dance is successful, a `Successfully retrieved access token` message shows up on the browser, which means the connector is now fully authorized and ready to use your Gmail account.
```

Comment:
"Mule runtime directs you..." to instead of "You are now redirected to", if it is Mule runtime doing this action.

Reply:
will verify with dev